### DISCUSSION TITLE
Brak collation pl_PL.UTF-8 w Azure Database for PostgreSQL

### DISCUSSION THREAD

**Discussion by 0utsider** [AzureCommunityPL/group#285] [Pomoc]
Nie wiem jak to ujac - ale chyba tak, ze jest to jawna dyskryminacja !
Chyba ze ja nie dowidze, a collaction pl jest ?!

``` 
xxx=> SELECT count(*)
FROM pg_collation;
 count
-------
  2055
(1 row)


xxx=> SELECT *
FROM pg_collation where collname like '%PL%';
 collname | collnamespace | collowner | collprovider | collencoding | collcollate | collctype | collversion
----------+---------------+-----------+--------------+--------------+-------------+-----------+-------------
(0 rows)
```

**Comment by mgrabarz** [comment]
Wydaje mi się, że format collation pl_PL.UTF-8 jest typowo Linuxowy. 

Wiem, że mało komu przychodzi do głowy by odpalać postgresa na Windowsie, ale to właśnie ma miejsce w Azure Database for Postgresql. Poszukaj Polish/Poland w like.

**Reply by 0utsider** [reply]
Niestety 

```
edb=> SELECT *
FROM pg_collation where collname ilike '%pol%';
 collname | collnamespace | collowner | collprovider | collencoding | collcollate | collctype | collversion
----------+---------------+-----------+--------------+--------------+-------------+-----------+-------------
(0 rows)
```

**Reply by smereczynski** [reply]
@mgrabarz Flexible już nie jest na Windows ;) Citus też :)

**Comment by 0utsider** [comment]
Zapytam inaczej - gdzie komu mozna zglosic ten jakby nie patrzec bug  w usludze ?

**Reply by miqm** [reply]
Najlepiej przez Support plan lub swojemu CSP.

**Reply by 0utsider** [reply]
Polecialo przez support - do ktorych swiat sie doczekam rozwiazania glupiego ? :)

**Reply by miqm** [reply]
Pierwszy kontakt raczej szybciej bo masz SLA

**Reply by 0utsider** [reply]
Niech ich jasny gwint, ciemna sroba walnie tak mi rozwiazali sprawe - https://developercommunity.visualstudio.com/t/missing-collation-pl-plutf-8-in-azure-database-for/1429277
Czyt olewczy poziom master. 

Tak wiec jakies inne pomysly ?

**Comment by piotrgwiazda** [comment]
Co ustawiasz przy tworzeniu bazy?

Wklej im że "initdb populates the system catalog pg_collation with
collations based on all the locales it finds on the operating system at the
time." ze strony https://www.postgresql.org/docs/current/collation.html

A tak poza tym to sprawdź jak działa und-x-icu, tylko tego collate nie da
się ustawić jako default.

wt., 25 maj 2021, 02:30 użytkownik 0utsider ***@***.***>
napisał:

**Reply by 0utsider** [reply]
```
XXX=> CREATE DATABASE new_database2
WITH TEMPLATE = template0
ENCODING = 'UTF8'
LC_COLLATE = 'pl_PL.UTF-8'
LC_CTYPE = 'pl_PL.UTF-8';
ERROR:  invalid locale name: "pl_PL.UTF-8"
```

Ja mam support microsoftu pouczac odnosnie ich wlasnego produktu - to na czubek malej glowki mi taki support z ich strony :( Napisze - zobaczymy co odpowiedza - bo to jakas kpina jest - jak zwykle korpo traktuje klienta templatem.

**Comment by piotrgwiazda** [comment]
Myślałem o "az postgres db create" lub jakimś odpowiedniku w Terraform. Nie
wiem czy to jest równoważne z CREATE DATABASE za pomocą DDL.

Support to support, dąży do zamknięcia ticketu. Ta ścieżka dotyczy każdego
produktu SaaS i znam znacznie gorsze supporty ;-)

wt., 25 maj 2021, 11:00 użytkownik 0utsider ***@***.***>
napisał:

**Reply by 0utsider** [reply]
Przez az tez sprawdzałem - podobny blad z info o braku takich locale

**Comment by piotrgwiazda** [comment]
Zostaje przetestować und-x-icu

pt., 28 maj 2021, 08:56 użytkownik 0utsider ***@***.***>
napisał:
